apps/deploy_monitoring/src/monitors/resource.rs

```diff
@@ -0,0 +1,204 @@
+// Copyright (c) SimpleStaking and Tezedge Contributors
+// SPDX-License-Identifier: MIT
+#![forbid(unsafe_code)]
+
+use crate::monitors::Alerts;
+use crate::node::OcamlNode;
+use crate::slack::SlackServer;
+use chrono::Utc;
+
+use serde::Serialize;
+use slog::Logger;
+use std::collections::VecDeque;
+use std::sync::{Arc, RwLock};
+
+use getset::Getters;
+use sysinfo::{System, SystemExt};
+
+use shell::stats::memory::ProcessMemoryStats;
+
+use crate::display_info::DiskData;
+use crate::node::{Node, TezedgeNode, OCAML_PORT, TEZEDGE_PORT};
+
+pub type ResourceUtilizationStorage = Arc<RwLock<VecDeque<ResourceUtilization>>>;
+
+/// The max capacity of the VecDeque holding the measurements
+pub const MEASUREMENTS_MAX_CAPACITY: usize = 40320;
+
+pub struct ResourceMonitor {
+    ocaml_resource_utilization: ResourceUtilizationStorage,
```

Comment:
@adonagy 
change here to `resource_utilization: Vec<ResourceUtilizationStorage>`